### PR TITLE
feat: Add buffering to provisioner job logs

### PR DESCRIPTION
**Pull request by kylecarbs**
This should improve overall build performance, and especially under load.

It removes the old `id` column on the `provisioner_job_logs` table and replaces it with an auto-incrementing big integer to preserve order.

Funny enough, we never had to care about order before because inserts would at minimum be 1ms different. Now they aren't, so the order needs to be preserved.

### CODE REVIEW

File: coderd/database/migrations/000071_provisioner_log_lines.up.sql [hunk 1]
```diff
@@ -0,0 +1,3 @@
+ALTER TABLE provisioner_job_logs DROP COLUMN id;
+
+ALTER TABLE provisioner_job_logs ADD COLUMN id BIGSERIAL PRIMARY KEY;
```
Comment (coadler) [inline coderd/database/migrations/000071_provisioner_log_lines.up.sql:3]: Can we guarantee that adding this bigserial column will keep the same order of logs?

**Review by coadler — APPROVED** [review]

File: provisionerd/runner/runner.go [hunk 2]
```diff
@@ -945,3 +897,41 @@ func (r *Runner) startTrace(ctx context.Context, name string, opts ...trace.Span
...
+		r.flushQueuedLogs(ctx)
+	})
+}
+
+func (r *Runner) flushQueuedLogs(ctx context.Context) {
+	r.mutex.Lock()
+	if r.flushLogsTimer != nil {
+		r.flushLogsTimer.Stop()
+	}
+	logs := r.queuedLogs[:]
```
[the remark this answers is not in the document]
Comment (kylecarbs) [reply provisionerd/runner/runner.go:924]: Yup, changing

File: coderd/database/migrations/000071_provisioner_log_lines.up.sql (hunk 1 above)
Comment (kylecarbs) [reply coderd/database/migrations/000071_provisioner_log_lines.up.sql:3]: It likely won't in all cases, but it seems to in most.

Having a _safe_ migration seems unreasonable here, so we settle for pretty great.

File: provisionerd/runner/runner.go [hunk 3]
```diff
@@ -945,3 +897,41 @@ func (r *Runner) startTrace(ctx context.Context, name string, opts ...trace.Span
...
+
+func (r *Runner) queueLog(ctx context.Context, log *proto.Log) {
+	r.mutex.Lock()
+	defer r.mutex.Unlock()
+	r.queuedLogs = append(r.queuedLogs, log)
+	if r.flushLogsTimer != nil {
+		r.flushLogsTimer.Reset(r.logBufferInterval)
+		return
+	}
+	if len(r.queuedLogs) > 100 {
```
[the remark this answers is not in the document]
Comment (kylecarbs) [reply provisionerd/runner/runner.go:909]: I added a comment and left it reconfigurable for now.

**Pull request merged**